packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -493,22 +506,124 @@ class _RawAutocompleteState<T extends Object> extends State<RawAutocomplete<T>>
       controller: _optionsViewController,
       overlayChildBuilder: _buildOptionsView,
       child: TextFieldTapRegion(
-        child: Container(
-          key: _fieldKey,
-          child: Shortcuts(
-            shortcuts: _shortcuts,
-            child: Actions(
-              actions: _actionMap,
-              child: CompositedTransformTarget(
-                link: _optionsLayerLink,
+        child: LayoutBuilder(
+          builder: (BuildContext context, BoxConstraints boxConstraints) {
+            assert(boxConstraints.hasBoundedWidth);
+            SchedulerBinding.instance.addPostFrameCallback((Duration duration) {
+              _fieldBoxConstraints.value = boxConstraints;
+            });
+            return Shortcuts(
+              shortcuts: _shortcuts,
+              child: Actions(
+                key: _fieldKey,
+                actions: _actionMap,
                 child: fieldView,
               ),
-            ),
-          ),
+            );
+          },
+        ),
+      ),
+    );
+  }
+}
+
+/// Positions the options view.
+class _OptionsLayoutDelegate extends SingleChildLayoutDelegate {
+  _OptionsLayoutDelegate({
+    required this.fieldBoxConstraints,
+    required this.fieldKey,
+    required Size? fieldSize,
+    required this.optionsViewOpenDirection,
+    required this.overlayContext,
+    required TextDirection? textDirection,
+  }) : fieldSize = fieldSize ?? Size.zero,
+       textDirection = textDirection ?? TextDirection.ltr;
+
+  /// The constraints of the field in [RawAutocomplete.fieldViewBuilder].
+  final BoxConstraints fieldBoxConstraints;
```

Comment:
Why are the constraints still needed if you know the size of the field?

Reply:
You're right it's probably better to just make these constraints match the width of the field rather than the field's width constraints, I'll do that.